### PR TITLE
Ensure the go vendored code is up-to-date and that we actually can call `cargo vendor` on every pull-request

### PR DESCRIPTION
**Pull request by fidencio**
The idea behind this commit is to never again release a tarball that would not be able to be consumed by downstream maintainers.

Fixes: #2159

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,4 +29,10 @@ $(eval $(call create_all_rules,$(COMPONENTS),$(TOOLS),$(STANDARD_TARGETS)))
 generate-protocols:
 	make -C src/agent generate-protocols
 
+check-vendor:
+	make -C src/runtime check-vendor
+	make -C src/agent vendor
+	make -C src/trace-forwarder vendor
+	make -C tools/agent-ctl vendor
+
```
Comment (jodh-intel) [inline Makefile:37]: I guess this is fine, but we can "fold" these specific rules into the auto-generated rules relatively easily and cleanly. Something like:

```diff
diff --git a/Makefile b/Makefile
index b3496bd7..f536c9bc 100644
--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ TOOLS =
 
 TOOLS += agent-ctl
 
-STANDARD_TARGETS = build check clean install test
+STANDARD_TARGETS = build check clean install test vendor
 
 include utils.mk
 
@@ -29,10 +29,4 @@ $(eval $(call create_all_rules,$(COMPONENTS),$(TOOLS),$(STANDARD_TARGETS)))
 generate-protocols:
 	make -C src/agent generate-protocols
 
-check-vendor:
-	make -C src/runtime check-vendor
-	make -C src/agent vendor
-	make -C src/trace-forwarder vendor
-	make -C tools/agent-ctl vendor
-
 .PHONY: all default
diff --git a/src/runtime/Makefile b/src/runtime/Makefile
index 05433505..536db828 100644
--- a/src/runtime/Makefile
+++ b/src/runtime/Makefile
@@ -648,12 +648,12 @@ install-scripts: $(SCRIPTS)
 install-completions:
 	$(QUIET_INST)install --mode 0644 -D  $(BASH_COMPLETIONS) $(DESTDIR)/$(BASH_COMPLETIONSDIR)/$(notdir $(BASH_COMPLETIONS));
 
-vendor:
+handle-vendor:
 	go mod tidy
 	go mod vendor
 	go mod verify
 
-check-vendor: vendor
+vendor: handle-vendor
 	./hack/tree_status.sh
 
 clean:
diff --git a/utils.mk b/utils.mk
index 3ad884fe..3435dd73 100644
--- a/utils.mk
+++ b/utils.mk
@@ -13,9 +13,9 @@
 # $1 - Directory component lives in.
 # $2 - Name of component.
 #
-# Note: The "clean" rule is the "odd one out" - it only depends on the
-# Makefile. This ensure that running clean won't first try to build the
-# project.
+# Note: The "clean" and "vendor" rules are the "odd ones out" - they only
+# depends on the Makefile. This ensure that running them won't first try to
+# build the project.
 
 define make_rules
 $(2) : $(1)/$(2)/Makefile
@@ -25,6 +25,9 @@ build-$(2) : $(2)
 check-$(2) : $(2)
 	make -C $(1)/$(2) check
 
+vendor-$(2) : $(1)/$(2)/Makefile
+	make -C $(1)/$(2) vendor
+
 clean-$(2) : $(1)/$(2)/Makefile
 	make -C $(1)/$(2) clean
 
@@ -40,6 +43,7 @@ test-$(2) : $(2)
     clean-$(2) \
     check-$(2) \
     test-$(2) \
+    vendor=$(2) \
     install-$(2)
 endef
 
```

Comment (jodh-intel) [reply Makefile:37]: That keeps the top-level `Makefile` clean and (hopefully) relatively clear.

Comment (fidencio) [reply Makefile:37]: @jodh-intel, I'm absolutely find on following your suggestions, I was just not sure whether we want `vendor` to be part of the auto-generated rules.

I'll update the PR soon, after I get a green run from the static checks.

Comment (fidencio) [reply Makefile]: @jodh-intel, I've addressed your comments and I'm marking this as resolved.
Please, if you spot something that I may have missed, please, feel free to mark it as unresolved.

Thanks for the suggestion!

**Comment by fidencio** [comment]
For some reason it's not that easy to retrigger the status checks without having to close the PR and reopen it

**Review by liubin — APPROVED** [review]

**Review by jodh-intel — APPROVED** [review]
Thanks @fidencio !

lgtm

**Pull request merged**